lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -435,15 +497,25 @@ assignMemoryLayouts(llvm::SmallVector<std::tuple<Operation *, int, Operation *>>
     }
 
     if (use->hasTrait<OpTrait::DotLike>()) {
+      auto mmaLoadType = getMMALoadType(op);
+      auto dot = dyn_cast<tt::DotOp>(use);
+      auto warpGroupDot = dyn_cast<ttng::WarpGroupDotOp>(use);
+
       loadInfo.usedByDot = true;
-      if (loadIsMMAv3(op)) {
-        loadInfo.loadIsMMAV3 = true;
+      loadInfo.isMMAv3Shared = mmaLoadType == MMALoadType::SharedV3;
+      loadInfo.isMMAv3Registers = (mmaLoadType == MMALoadType::Registers)
+        && warpGroupDot;
+
+      if (loadInfo.isMMAv3Shared) {
         loadInfo.sharedEncoding =
             getSharedEncoding(op, /*loadIsMMAv3=*/true).value_or(nullptr);
       } else if (isa<tt::ExperimentalDescriptorLoadOp>(op)) {
         loadInfo.sharedEncoding =
             getSharedEncoding(op, /*loadIsMMAv3=*/true).value_or(nullptr);
-      } else if (auto dot = dyn_cast<tt::DotOp>(use)) {
+      } else if (loadInfo.isMMAv3Registers || dot) {
+        // if warpGroupDot, we must now have operand A in registers since
+        // loadIsMMAv3Shared is false from above if-check
```

Comment:
does this commend make sense anymore? it seems it's specifically in registers because we checked that it's specifically in registers, not just because it's not mmav3shared.

Reply:
I forgot to remove this, thanks for the reminder. Removed in https://github.com/openxla/triton/pull/19.